src/main/java/org/opentripplanner/transit/model/timetable/ScheduledTripTimes.java

```diff
@@ -53,18 +53,28 @@ public final class ScheduledTripTimes implements TripTimes {
   private final List<BookingInfo> dropOffBookingInfos;
   private final List<BookingInfo> pickupBookingInfos;
 
+  /**
+   * The headsign displayed by the vehicle at each stop along this trip. The array index is the
+   * stop index within this trip's TripPattern. Any number of array elements may point to the same
+   * I18NString instance if the headsign remains unchanged between stops.
+   */
   @Nullable
   private final I18NString[] headsigns;
 
   /**
-   * Implementation notes: This is 2D array since there can be more than one via name/stop per each
-   * record in stop sequence). Outer array may be null if there are no vias in stop sequence. Inner
-   * array may be null if there are no vias for particular stop. This is done in order to save
-   * space.
+   * A 2D array of String containing zero or more Via messages displayed at each stop in the
+   * stop sequence. A Via is an additional intermediate destination that is displayed alongside the
+   * terminus headsign, but will usually change or only be displayed at certain stops along the way.
+   * While the concept of Headsigns exists in both GTFS (Headsign) and Netex (DestinationDisplay),
+   * the Via concept is only present in Transmodel. This reference be null if no stop in the entire
+   * sequence of stops has any via strings. Any subarray may also be null or empty if no Via strings
+   * are displayed at that particular stop. These nulls are allowed to conserve memory in the common
+   * case where there are few or no via messages.
```

Comment:
I moved that information to the getter ScheduledTripTimes#getHeadsignVias. The question then arises how far up the chain of accessors the information should be placed, and whether it is acceptable for it to be replicated.  

It turns out similar information was already available on RealTimeTripTimes#getHeadsignVias and in the interface declaration TripTimes#getHeadsignVias, but I somehow didn't spot or understand these when working on ScheduledTripTimes. No such explanation is present on TripTimeOnDate#getHeadsignVias which reads through to the same data.

In the end I consolidated information from the various Javadoc on the interface (overridden) method declaration, and replicated it only on TripTimeOnDate#getHeadsignVias because it's not readily reachable from an overridden method.